builder/src/lib.rs

```diff
@@ -0,0 +1,732 @@
+    consensus::ConsensusMetricsValue,
+    constants::{Version01, STATIC_VER_0_1},
+    event::Event,
+    light_client::StateKeyPair,
+    signature_key::{BLSPrivKey, BLSPubKey},
+    traits::election::Membership,
+    traits::metrics::Metrics,
+    HotShotConfig, PeerConfig, ValidatorConfig,
+};
+use std::fmt::Display;
+// Should move `STAKE_TABLE_CAPACITY` in the sequencer repo when we have variate stake table support
+use hotshot_stake_table::config::STAKE_TABLE_CAPACITY;
+
+use async_std::sync::{Arc, RwLock};
+use async_std::task::{spawn, JoinHandle};
+
+use async_compatibility_layer::art::{async_sleep, async_spawn};
+use hotshot_builder_api::builder::{
+    BuildError, Error as BuilderApiError, Options as HotshotBuilderApiOptions,
+};
+use hotshot_builder_core::service::GlobalState;
+use hotshot_state_prover;
+use jf_primitives::{
+    merkle_tree::{namespaced_merkle_tree::NamespacedMerkleTreeScheme, MerkleTreeScheme},
+    signatures::bls_over_bn254::VerKey,
+};
+use sequencer::catchup::mock::MockStateCatchup;
+use sequencer::state_signature::StakeTableCommitmentType;
+use sequencer::{
+    catchup::StatePeers,
+    context::{Consensus, SequencerContext},
+    l1_client::L1Client,
+    network,
+    persistence::SequencerPersistence,
+    state::FeeAccount,
+    state::ValidatedState,
+    state_signature::{static_stake_table_commitment, StateSigner},
+    BuilderParams, L1Params, NetworkParams, Node, NodeState, PrivKey, PubKey, SeqTypes,
+};
+use std::{alloc::System, any, fmt::Debug, mem};
+use std::{marker::PhantomData, net::IpAddr};
+use std::{net::Ipv4Addr, thread::Builder};
+use tide_disco::{app, method::ReadState, App, Url};
+use versioned_binary_serialization::version::StaticVersionType;
+
+pub mod non_permissioned;
+pub mod permissioned;
+
+// It runs the api service for the builder
+pub fn run_builder_api_service(url: Url, source: Arc<RwLock<GlobalState<SeqTypes>>>) {
+    // it is to serve hotshot
+    let builder_api = hotshot_builder_api::builder::define_api::<
+        Arc<RwLock<GlobalState<SeqTypes>>>,
+        SeqTypes,
+        Version01,
+    >(&HotshotBuilderApiOptions::default())
+    .expect("Failed to construct the builder APIs");
+
+    // it enables external clients to submit txn to the builder's private mempool
+    let private_mempool_api = hotshot_builder_api::builder::submit_api::<
+        Arc<RwLock<GlobalState<SeqTypes>>>,
+        SeqTypes,
+        Version01,
+    >(&HotshotBuilderApiOptions::default())
+    .expect("Failed to construct the builder API for private mempool txns");
+
+    let mut app: App<Arc<RwLock<GlobalState<SeqTypes>>>, BuilderApiError, Version01> =
+        App::with_state(source);
+
+    app.register_module("block_info", builder_api)
+        .expect("Failed to register the builder API");
+
+    app.register_module("txn_submit", private_mempool_api)
+        .expect("Failed to register the private mempool API");
+
+    async_spawn(app.serve(url, STATIC_VER_0_1));
+}
+
+#[cfg(test)]
```

Comment:
A lot of this feels like it belongs in non_permissioned.rs or permissioned.rs

It's okay, the non-test contents of lib.rs fall within the reasonable terseness, but it makes the file look huge on first glance.